proto-wine/proton.cpp

```diff
@@ -0,0 +1,64 @@
+/* Copyright (c) 2019 Stanislaw Halik <sthalik@misaki.pl>
+ *
+ * Permission to use, copy, modify, and/or distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ */
+
+#ifndef OTR_WINE_NO_WRAPPER
+
+#include <QtGlobal>
+#include <QString>
+#include <QProcessEnvironment>
+
+QProcessEnvironment make_steam_environ(const QString& proton_version)
+{
+    auto ret = QProcessEnvironment::systemEnvironment();
+    QString home = qgetenv("HOME");
+
+    auto expand = [&](QString x) {
+        x.replace("HOME", home);
+        x.replace("PROTON", proton_version);
+        return x;
+    };
+
+    QString path = expand(
+        "HOME/.local/share/Steam/steamapps/common/Proton PROTON/dist/bin:"
+        "HOME/.local/share/Steam/ubuntu12_32/steam-runtime/amd64/bin:"
+        "HOME/.local/share/Steam/ubuntu12_32/steam-runtime/amd64/usr/bin"
+    );
+    path += ':'; path += qgetenv("PATH");
+    ret.insert("PATH", path);
+
+    QString library_path = expand(
+        ":HOME/.local/share/Steam/steamapps/common/Proton PROTON/dist/lib"
+        ":HOME/.local/share/Steam/steamapps/common/Proton PROTON/dist/lib64"
+        ":HOME/.local/share/Steam/ubuntu12_32/steam-runtime/pinned_libs_32"
+        ":HOME/.local/share/Steam/ubuntu12_32/steam-runtime/pinned_libs_64"
+        ":HOME/.local/share/Steam/ubuntu12_32/steam-runtime/i386/lib/i386-linux-gnu"
+        ":HOME/.local/share/Steam/ubuntu12_32/steam-runtime/i386/lib"
+        ":HOME/.local/share/Steam/ubuntu12_32/steam-runtime/i386/usr/lib/i386-linux-gnu"
+        ":HOME/.local/share/Steam/ubuntu12_32/steam-runtime/i386/usr/lib"
+        ":HOME/.local/share/Steam/ubuntu12_32/steam-runtime/amd64/lib/x86_64-linux-gnu"
+        ":HOME/.local/share/Steam/ubuntu12_32/steam-runtime/amd64/lib"
+        ":HOME/.local/share/Steam/ubuntu12_32/steam-runtime/amd64/usr/lib/x86_64-linux-gnu"
+        ":HOME/.local/share/Steam/ubuntu12_32/steam-runtime/amd64/usr/lib"
+    );
+    library_path += ':'; library_path += qgetenv("LD_LIBRARY_PATH");
+    ret.insert("LD_LIBRARY_PATH", library_path);
+
+    ret.insert("WINEPREFIX", expand("HOME/.local/share/Steam/steamapps/compatdata/805550/pfx"));
+    ret.insert("WINEESYNC", "1");
```

Comment:
The WINEESYNC environment variable may differ per game, "1" is the default, but some games require ESYNC to be turned off. Within Steam a user can set this by e.g. the game launch options.
A user would need the possibility to override this one, either by UI, or by setting the same environment variable before starting opentrack. The same would apply to WINEFSYNC.
See https://github.com/ValveSoftware/Proton/blob/proton_4.11/proton#L518 and https://github.com/ValveSoftware/Proton/blob/proton_4.11/proton#L518